core/services/wifi/WifiManager.py

```diff
@@ -79,74 +79,74 @@ def __dict_from_list(data: bytes) -> Dict[str, Any]:
 
         return output
 
-    def get_wifi_available(self) -> List[ScannedWifiNetwork]:
+    async def get_wifi_available(self) -> List[ScannedWifiNetwork]:
         """Get a dict from the wifi signals available"""
         try:
-            self.wpa.send_command_scan(timeout=15)
-            data = self.wpa.send_command_scan_results()
+            await self.wpa.send_command_scan(timeout=15)
+            data = await self.wpa.send_command_scan_results()
             networks_list = WifiManager.__dict_from_table(data)
             return [ScannedWifiNetwork(**network) for network in networks_list]
         except Exception as error:
             raise FetchError("Failed to fetch wifi list.") from error
 
-    def get_saved_wifi_network(self) -> List[SavedWifiNetwork]:
+    async def get_saved_wifi_network(self) -> List[SavedWifiNetwork]:
         """Get a list of saved wifi networks"""
         try:
-            data = self.wpa.send_command_list_networks()
+            data = await self.wpa.send_command_list_networks()
             networks_list = WifiManager.__dict_from_table(data)
             return [SavedWifiNetwork(**network) for network in networks_list]
         except Exception as error:
             raise FetchError("Failed to fetch saved networks list.") from error
 
-    def set_wifi_password(self, credentials: WifiCredentials) -> Any:
+    async def set_wifi_password(self, credentials: WifiCredentials) -> Any:
         """Set network ssid and password
 
         Arguments:
             credentials {[WifiCredentials]} -- object containing ssid and password of the network
         """
         try:
-            data = self.wpa.send_command_add_network()
+            data = await self.wpa.send_command_add_network()
             data = data.strip()
             if data == b"Fail":
                 raise RuntimeError("Failed to add new network.")
 
             network_number = data.decode("utf-8")
-            self.wpa.send_command_set_network(network_number, "ssid", '"{}"'.format(credentials.ssid))
-            self.wpa.send_command_set_network(network_number, "psk", '"{}"'.format(credentials.password))
-            self.wpa.send_command_enable_network(network_number)
-            self.wpa.send_command_reconnect()
+            await self.wpa.send_command_set_network(str(network_number), "ssid", '"{}"'.format(credentials.ssid))
```

Comment:
Why network_number should needs a cast with await ?

Reply:
It doesn't need, it was a rebase mistake. Will fix. (still, this is a middle change. the final version refactors this part)

Reply:
Fixed.